File: sample/dist2/java/src/main/java/org/yanex/hello/IBaseInterface.java
package org.yanex.hello;

@javax.annotation.Generated(value = "jsii-pacmak/0.19.0 (build 5d7824d)", date = "2019-10-28T18:25:08.409Z")
public interface IBaseInterface extends software.amazon.jsii.JsiiSerializable, org.yanex.hello.IVeryBaseInterface {

    void bar();

    /**
     * A proxy class which represents a concrete javascript instance of this type.
     */
    final static class Jsii$Proxy extends software.amazon.jsii.JsiiObject implements org.yanex.hello.IBaseInterface {
        protected Jsii$Proxy(final software.amazon.jsii.JsiiObjectRef objRef) {
            super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
            this.setObjRef(objRef);
        }

        @Override
        public void bar() {
            this.jsiiCall("bar", Void.class);
        }

        @Override
        public void foo() {
            this.jsiiCall("foo", Void.class);
        }
    }
}

File: sample/dist2/kotlin/src/main/kotlin/hello/IBaseInterface.kt
package org.yanex.hello

@javax.annotation.Generated(value = "jsii-pacmak/0.19.0 (build 5d7824d)", date = "2019-10-29T17:33:08.481Z")
interface IBaseInterface: software.amazon.jsii.JsiiSerializable, org.yanex.hello.IVeryBaseInterface {
    fun bar() {
        this.jsiiCall("bar", kotlin.Any::class.java)
    }

    internal class `Jsii$Proxy`: software.amazon.jsii.JsiiObject, org.yanex.hello.IBaseInterface {
        protected constructor(objRef: software.amazon.jsii.JsiiObjectRef) : super(software.amazon.jsii.JsiiObject.InitializationMode.JSII) {
            this.objRef = objRef
        }

        fun foo() {
            this.jsiiCall("foo", kotlin.Any::class.java)
        }

        fun bar() {
            this.jsiiCall("bar", kotlin.Any::class.java)
        }
    }
}
